src/controller/python/chip/crypto/p256keypair.py

```diff
@@ -0,0 +1,71 @@
+from typing import *
+from ctypes import *
+
+from ecdsa import NIST256p, SigningKey, ECDH
+import abc
+
+from chip import native
+
+_pychip_ECDSA_sign_msg_func = CFUNCTYPE(c_bool, py_object, c_char_p, c_size_t, c_char_p, POINTER(c_uint32))
+
+_pychip_ECDH_derive_secret_func = CFUNCTYPE(c_bool, py_object, c_char_p, c_char_p, POINTER(c_uint32))
+
+P256_PUBLIC_KEY_LENGTH = 2 * 32 + 1
+
+
+@_pychip_ECDSA_sign_msg_func
+def _pychip_ECDSA_sign_msg(self_: 'P256Keypair', message_buf: c_char_p, message_size: int, signature_buf: c_char_p, signature_buf_size: POINTER(c_uint32)) -> bool:
+    res = self_.ECDSA_sign_msg(string_at(message_buf, message_size)[:])
+    memmove(signature_buf, res, len(res))
+    signature_buf_size.content = len(res)
+    return True
+
+
+@_pychip_ECDH_derive_secret_func
+def _pychip_ECDH_derive_secret(self_: 'P256Keypair', remote_pubkey: c_char_p, out_secret_buf: c_char_p, out_secret_buf_size: POINTER(c_uint32)) -> bool:
+    res = self_.ECDH_derive_secret(string_at(remote_pubkey, P256_PUBLIC_KEY_LENGTH)[:])
+    memmove(out_secret_buf, res, len(res))
+    out_secret_buf_size.content = len(res)
+    return True
+
+
+class P256Keypair:
+    def __init__(self):
+        pass
+
+    def create_native_object(self) -> c_void_p:
+        handle = native.GetLibraryHandle()
+        if not handle.pychip_NewP256Keypair.argtypes:
+            setter = native.NativeLibraryHandleMethodArguments(handle)
+            setter.Set("pychip_NewP256Keypair", c_void_p, [py_object, c_char_p,
+                       _pychip_ECDSA_sign_msg_func, _pychip_ECDH_derive_secret_func])
+        self._native_obj = handle.pychip_NewP256Keypair(self, self._pubkey, _pychip_ECDSA_sign_msg, _pychip_ECDH_derive_secret)
+        return self._native_obj
```

Comment:
Please convert to a function (a builder, in essence). P256Keypair in Python should allow a full Python implementation.

Reply:
done